File: lab1/w01a_java/app/src/main/java/com/example/w01a/MainActivity.java
package com.example.w01a;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Random;

public class MainActivity extends AppCompatActivity {
    private int n;
    private int m;
    private int score;
    private Random generator;

    private TextView points;
    private Button leftButton;
    private Button rightButton;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        this.init();
        this.roll();
    }

    private void init () {
        this.score = 0;
        this.generator = new Random();

        this.points = findViewById(R.id.points);
        this.leftButton = findViewById(R.id.leftButton);
        this.rightButton = findViewById(R.id.rightButton);
    }

    private void roll() {
        this.points.setText("Punkty: " + this.score);

        this.n = this.generator.nextInt(10);
        this.m = this.generator.nextInt(10);
        this.leftButton.setText("" + this.n);
        this.rightButton.setText("" + this.m);
    }

    public void leftClick(View view) {
        if (this.n >= this.m) {
            this.score++;
            Toast.makeText(this, "Correct!", Toast.LENGTH_SHORT).show();
        }
        else {
            this.score--;
            Toast.makeText(this, "Incorrect!", Toast.LENGTH_SHORT).show();
        }

        this.roll();
    }

    public void rightClick(View view) {
        if (this.m >= this.n) {
            this.score++;
            Toast.makeText(this, "Correct!", Toast.LENGTH_SHORT).show();
        }
        else {
            this.score--;
            Toast.makeText(this, "Incorrect!", Toast.LENGTH_SHORT).show();
        }

        this.roll();
    }
}

File: lab1/w01a_kotlin/app/src/main/java/com/example/w01a/MainActivity.kt
package com.example.w01a

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView
import android.widget.Toast
import kotlin.random.Random

class MainActivity : AppCompatActivity() {
    private var n: Int = 0
    private var m: Int = 0
    private var score: Int = 0


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        this.score = 0
        this.roll()
    }

    fun roll () {
        findViewById<TextView>(R.id.points).text = "Punkty: " + this.score
        this.n = Random.nextInt(10)
        this.m = Random.nextInt(10)
        findViewById<TextView>(R.id.leftButton).text = "" + this.n
        findViewById<TextView>(R.id.rightButton).text = "" + this.m
    }

    fun leftClick(view: View) {
        if (this.n >= this.m) {
            this.score++
            Toast.makeText(this, "Correct!", Toast.LENGTH_SHORT).show()
        }
        else {
            this.score--
            Toast.makeText(this, "Incorrect!", Toast.LENGTH_SHORT).show()
        }

        this.roll()
    }

    fun rightClick(view: View) {
        if (this.m >= this.n) {
            this.score++
            Toast.makeText(this, "Correct!", Toast.LENGTH_SHORT).show()
        }
        else {
            this.score--
            Toast.makeText(this, "Incorrect!", Toast.LENGTH_SHORT).show()
        }

        this.roll()
    }
}